wiring/src/spark_wiring_ble.cpp

```diff
@@ -638,12 +643,14 @@ size_t BleAdvertisingData::append(BleAdvertisingDataType type, const uint8_t* bu
             selfData_.insert(offset + 2, buf, len);
         }
     }
-    else if ((selfData_.size() + len + 2) <= BLE_MAX_ADV_DATA_LEN_EXT) {
+    else if ((selfData_.size() + len + 2) <= maxAdvertisingDataLen_) {
         // Append the AD structure at the and of advertising data.
         CHECK_TRUE(selfData_.reserve(selfData_.size() + len + 2), selfData_.size());
         selfData_.append(len + 1);
         selfData_.append(static_cast<uint8_t>(type));
         selfData_.append(buf, len);
+    } else {
+        return SYSTEM_ERROR_TOO_LARGE;
```

Comment:
This still doesn't solve the problem of using HAL directly (which for example ble provisioning mode does). Let's make sure to return a proper error in there.

Reply:
Ill fix the direct HAL usage with a new platform define like `BLE_MAX_SUPPORTED_ADV_DATA_LEN`
Ill use this new define in wiring layer and internally in the HAL, plus wherever else its used like in provisioning.